File: Veterinario/VeterinarioBlueJ/DetalleReceta.java
public class DetalleReceta {
    private Medicamento medicamento;
    private int cantidad;

    public DetalleReceta(Medicamento medicamento, int cantidad) {
        this.medicamento = medicamento;
        this.cantidad = cantidad;
    }

    public Medicamento getMedicamento() { return medicamento; }
    public int getCantidad() { return cantidad; }
}

File: Veterinario/ConsultorioVeterinario-V2/ConsultorioVeterinario/Logica/DetalleReceta.cs
namespace ConsultorioVeterinario.Logica
{
    public class DetalleReceta
    {
        public Medicamento Medicamento { get; set; }
        public int Cantidad { get; set; }

        public DetalleReceta(Medicamento medicamento, int cantidad)
        {
            Medicamento = medicamento;
            Cantidad = cantidad;
        }
    }
}